source/application/AMPIC887DriverServer/AMPIC887ControllerState.cpp

```diff
@@ -0,0 +1,67 @@
+#include "AMPIC887ControllerState.h"
+
+AMPIC887ControllerState::AMPIC887ControllerState()
+{
+	hexapodState_ = new AMPIC887HexapodState();
+	dataRecorderState_ = new AMPIC887DataRecorderState();
+	currentCommandLevel_ = AMGCS2::UnknownCommandLevel;
+	isInitialized_ = false;
+}
+
+AMPIC887ControllerState::~AMPIC887ControllerState()
+{
+	delete hexapodState_;
+	delete dataRecorderState_;
+}
+
+AMPIC887HexapodState * AMPIC887ControllerState::hexapodState() const
+{
+	return hexapodState_;
+}
+
+AMPIC887DataRecorderState * AMPIC887ControllerState::dataRecorderState() const
+{
+	return dataRecorderState_;
+}
+
+void AMPIC887ControllerState::initialize(AMGCS2::ControllerCommandLevel commandLevel,
+										 const QString &availableParameters,
+										 const QString& identification)
+{
+	currentCommandLevel_ = commandLevel;
```

Comment:
Is it possible that these available parameters and identification can be not valid causing this to not actually become initialized?